test/commands.test.ts

```diff
@@ -26,61 +26,130 @@ import {Guid} from '../src/guid';
 import {GuidCommands} from '../src/commands';
 
 suite('GuidCommands', () => {
-    test('quick pick 1 is simple string', () => {
-        var g = Guid.parse('12341234-1234-1234-1234-123412341234');
-        var items = GuidCommands.getQuickPickItems(g);
+    test('quick pick items are correct when all settings are true', () => {
+        const g = Guid.parse('12341234-dead-beef-1234-123412341234');
+        const items = GuidCommands.getQuickPickItems(g, true, true, true);
+
+        assert.equal(items.length, 6);
+
+        const item1 = items[0];
+        assert.strictEqual(item1.label, '1');
+        assert.equal(item1.description, '12341234-dead-beef-1234-123412341234');
+        assert.equal(item1.text, '12341234-dead-beef-1234-123412341234');
+
+        const item2 = items[1];
+        assert.strictEqual(item2.label, '2');
+        assert.equal(item2.description, '{12341234-dead-beef-1234-123412341234}');
+        assert.equal(item2.text, '{12341234-dead-beef-1234-123412341234}');
 
-        assert.equal(items.length, 4);
+        const item3 = items[2];
+        assert.strictEqual(item3.label, '3');
+        assert.equal(item3.description, '12341234-DEAD-BEEF-1234-123412341234');
+        assert.equal(item3.text, '12341234-DEAD-BEEF-1234-123412341234');
 
-        var item = items[0];
-        assert.strictEqual(item.label, '1');
-        assert.equal(item.description, '12341234-1234-1234-1234-123412341234');
-        assert.equal(item.text, '12341234-1234-1234-1234-123412341234');
+        const item4 = items[3];
+        assert.strictEqual(item4.label, '4');
+        assert.equal(item4.description, '{12341234-DEAD-BEEF-1234-123412341234}');
+        assert.equal(item4.text, '{12341234-DEAD-BEEF-1234-123412341234}');
+
+        const item5 = items[4];
+        assert.strictEqual(item5.label, '5');
+        assert.equal(item5.description,
+            'static const struct GUID __NAME__ = {0x12341234, 0xdead, 0xbeef, {0x12, 0x34, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34}};'
+        );
+        assert.equal(item5.text,
+            '// {12341234-1234-1234-1234-123412341234}\n' +
+            'static const struct GUID __NAME__ = {0x12341234, 0xdead, 0xbeef, {0x12, 0x34, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34}};\n'
+        );
+
+        const item6 = items[5];
+        assert.strictEqual(item5.label, '6');
+        assert.equal(item6.description,
+            'DEFINE_GUID(__NAME__, 0x12341234, 0xdead, 0xbeef, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34);'
+        );
+        assert.equal(item6.text,
+            '// {12341234-1234-1234-1234-123412341234}\n' +
+            'DEFINE_GUID(__NAME__, 0x12341234, 0xdead, 0xbeef, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34, 0x12, 0x34);\n'
+        );
     });
 
-    test('quick pick 2 is registry string', () => {
```

Comment:
I'm not sure how to do a full test on this.  I set the defaults [here](https://github.com/101100/vscode-guid/blob/add-uppercase-option/package.json#L21-L37) to match the old behaviour, but I don't know how to create an integration test to show that the defaults have the correct behaviour.